### PR TITLE
Add instructions to edit premade designs to scripting README files.

### PR DESCRIPTION
**Pull request by LGM-Doyle**

### CODE REVIEW

File: default/scripting/monster_designs/README.md [hunk 1]
```diff
@@ -0,0 +1,12 @@
+# Instructions for editing ship or monster designs in game in the Design window.
+1. Move or backup the `shipdesigns` directory in your freeorion data directory.
```
Comment (geoffthemedio) [inline default/scripting/monster_designs/README.md]: where is the "freeorion data directory" ?

File: default/scripting/monster_designs/README.md [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+2. Copy the `default/scripting/ship_designs` or `default/scripting/monsters` directory into your
+data directory and replace your `shipdesigns` directory.
```
Comment (geoffthemedio) [inline default/scripting/monster_designs/README.md]: "and replace", or "replacing" or "in place of" ?

File: default/scripting/monster_designs/README.md [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+3. Start a new game.
+4. In the Design windo, your saved ship designs will be the default ship designs or monsters.  Edit the ships.
+    i. Add new saved designs.
+    ii. Drag and drop to change ordering.
+    iii. If you use a stringtable index for **both** the ship name **and** description then it will save the stringtable index and use the translated text for display.  You can check this in the pedia window.  When you add/remove parts the name and description in the pedia window will be pulled from the stringtable.
```
Comment (geoffthemedio) [inline default/scripting/monster_designs/README.md]: explain how to use a stringtable index?

File: default/scripting/monster_designs/README.md [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+2. Copy the `default/scripting/ship_designs` or `default/scripting/monsters` directory into your
+data directory and replace your `shipdesigns` directory.
+3. Start a new game.
+4. In the Design windo, your saved ship designs will be the default ship designs or monsters.  Edit the ships.
+    i. Add new saved designs.
+    ii. Drag and drop to change ordering.
+    iii. If you use a stringtable index for **both** the ship name **and** description then it will save the stringtable index and use the translated text for display.  You can check this in the pedia window.  When you add/remove parts the name and description in the pedia window will be pulled from the stringtable. 
+    iv. Note: You can not start a ship with a monster hull, but you can add/remove components to monster designs in the saved design directory.
+5. Exit the game.
+6. Copy the `shipdesigns` directory from your data directory back into the `default/scripting/ship_designs` or `default/scripting/monsters` directory.
```
Comment (geoffthemedio) [inline default/scripting/monster_designs/README.md]: does one need to copy the directory, or just its contents?

**Review by LGM-Doyle — COMMENTED** [review]
Changes made.

File: default/scripting/monster_designs/README.md (hunk 4 above)
Comment (geoffthemedio) [reply default/scripting/monster_designs/README.md]: Because someone reading it may be confused about what they need to do.

**Comment by Vezzra** [comment]
As these additions are essentially documentation, this should either be tagged as "user manual" or "technical docs". Not 100% sure which applies here, for now I'm going with "technical docs".

Any opinons/objections? @geoffthemedio, @LGM-Doyle, @adrianbroher?

**Comment by MatGB** [comment]
I thought user manual was in game Pedia and technical docs was anything else but especially how to change things but I might be wrong.

**Comment by LGM-Doyle** [comment]
"technical docs" seems reasonable to me.

File: README.md [hunk 5]
```diff
@@ -59,6 +59,30 @@ packages via the command line.
...
   * [ArchLinux package] stable release
 
+## Directories
+* install directory  
+The location of this README.md file, called `<install_dir>` below.  
+* config - game settings  
+    * linux  
+`$XDG_CONFIG_HOME/freeorion` which defaults to `~/.config/freeorion`  
+    * OSX  
+`Username/Library/Application Support/FreeOrion/`
```
Comment (dbenage-cx) [inline README.md]: Replace `Username` with `$HOME` (or as previous "...which defaults to...") https://github.com/freeorion/freeorion/blob/master/util/Directories.cpp#L87 (and same with data dir)

File: README.md [hunk 6]
```diff
@@ -59,6 +59,30 @@ packages via the command line.
...
+## Directories
+* install directory  
+The location of this README.md file, called `<install_dir>` below.  
+* config - game settings  
+    * linux  
+`$XDG_CONFIG_HOME/freeorion` which defaults to `~/.config/freeorion`  
+    * OSX  
+`Username/Library/Application Support/FreeOrion/`  
+    * Windows  
+`C:\Users\Username\AppData\Roaming\FreeOrion`
```
Comment (dbenage-cx) [inline README.md]: similar: `$APPDATA\FreeOrion` https://github.com/freeorion/freeorion/blob/master/util/Directories.cpp#L374 (and same with data dir)

File: README.md [hunk 7]
```diff
@@ -59,6 +59,30 @@ packages via the command line.
...
+`C:\Users\Username\AppData\Roaming\FreeOrion`  
+* data - local user data, saved games, log files  
+    * linux  
+`$XDG_DATA_HOME/freeorion` which defaults to `~/.config/freeorion`  
+    * OSX  
+`Username/Library/Application Support/FreeOrion/`  
+    * Windows  
+`C:\Users\Username\AppData\Roaming\FreeOrion`  
+* stringtables - translation indices for various languages  
+    `<install_dir>/default/stringtables/`
```
Comment (dbenage-cx) [inline README.md]: Possibly define as use `resource directory` for these, in event a packager relocates the default resource directory.

**Review by dbenage-cx — CHANGES_REQUESTED** [review]

File: README.md (hunk 5 above)
Comment (LGM-Doyle) [reply README.md]: Done.

File: README.md (hunk 6 above)
Comment (LGM-Doyle) [reply README.md]: Done

File: README.md (hunk 7 above)
Comment (LGM-Doyle) [reply README.md]: I added the resource directory with a comment that it can be relocated in `config.xml` or `persistent_config.xml`

**Review by LGM-Doyle — COMMENTED** [review]
I've fixed all of the issues. 

Thanks for the comment.

**Review by dbenage-cx — APPROVED** [review]

**Comment by MatGB** [comment]
Before I open an issue, playing a new game from last night compiled from master at a4f1548, the only thing I've done is follow these instructions to reorder the premade designs.

On loading, none of my empire designs are there to build, not originals, not saved designs, not premades. I've tried saving and reloading, and even designs I put in aren't there. However, if I try to create a new design it won't let me add a duplicate of something that ought to be there (eg a basic outpost ship with medium hull and outpost). This is beyond frustrating, could it be because I reorderd the premades or is it a bigger bug?

As a workaround I've put the important premades into my saved designs, I can 'add all saved designs' whenever I reload, but that shouldn't happen.

**Comment by LGM-Doyle** [comment]
@MatGB I ran into that same bug and fixed it yesterday in https://github.com/freeorion/freeorion/commit/f0d71bd8f82e6ec568de821c012b633409b26bda

I made changes late in the process of PR #1587 to support legacy loading, which broke regular loading.  I removed those changes.

**Pull request merged**